prov/sockets/src/sock_av.c

```diff
@@ -188,7 +192,19 @@ static int sock_check_table_in(struct sock_av *_av, struct sockaddr_in *addr,
 							context, i, FI_ENOMEM);
 						continue;
 					}
+					_av->idx_arr[_av->table_hdr->stored] = _av->table_hdr->stored;
+					_av->table_hdr = new_addr;
+					_av->table_hdr->size = new_count;
+					_av->table = (struct sock_av_addr *)
+						((char *)_av->table_hdr +
+						idx_arr_sz +
+						sizeof(struct sock_av_table_hdr));
```

Comment:
I didn't look at the first set of patches, but we're indented 6 levels here.  Breaking this into multiple functions or restructuring the code may improve readability.